SHARDING.md

```diff
@@ -0,0 +1,254 @@
+by updating the module reference of `sigstore`, all shards would be
+concurrently updated.
+
+To add a new shard, create a module under either
+`terraform/environments/staging/1-infrastructure/staging.tf` or
+`terraform/environments/production/1-infrastructure/production.tf`.
+Name the module `rekor-tiles-log-<year>-<revision>`, e.g.
+`rekor-tiles-log-2025-1`.
+
+<!--- TODO: Add code pointers for module example -->
+
+Update `general.auto.tfvars` to include the shard variable values:
+
+```
+rekor_tiles_shards = {
+  "log2025-1" = {
+    ...
+  },
+  "log2025-2" = {
+    ...
+  },
+  "log2026-1" = {
+    ...
+  }
+}
+```
+
+After merging, run `terraform plan` through the GitHub Actions workflow
+for [staging](https://github.com/sigstore/public-good-instance/actions/workflows/env-staging.yml)
+or [prod](https://github.com/sigstore/public-good-instance/actions/workflows/env-prod.yml).
+Confirm the expected resources are created, then run `terraform apply` through the workflow.
+
+## Creating Key Material
+
+The public-good instance uses a Tink keyset to sign checkpoints. Tink
+encrypts a signing key using a "key encryption key" managed by GCP KMS,
+and loads the key into memory on service startup.
+
+Tink is beneficial for reliability and cost reduction - KMS is now only
+a startup dependency rather than a runtime dependency, and signing happens
+exclusively in memory.
+
+To create a shard signing key:
+
+1. Grant yourself access to use the new signing key. Update the `staging.tf`
+or `production.tf` config to include:
+
+```
+resource "google_kms_key_ring_iam_member" "rekor-tiles-keyring" {
+  key_ring_id = "<project-rekor or projectsigstore-staging>/<location, e.g. global>/<name of shard keyring>"
+  role        = "roles/cloudkms.cryptoOperator"
+
+  member = "user:<username>@sigstore.dev"
+}
+```
+
+2. Run `terraform plan` and `terraform apply` using the GHA workflow.
+
+3. Using [tinkey](https://developers.google.com/tink/tinkey-overview#installation), create an encrypted Tink keyset:
+
+```
+tinkey create-keyset --key-template ED25519 --out enc-keyset.cfg --master-key-uri gcp-kms://projects/<project name>/locations/<region>/keyRings/<key-ring>/cryptoKeys/<key>
+```
+
+Save the output file `enc-keyset.cfg`, which will be used as a value in the Helm chart.
+
+4. Revoke IAM access by removing `google_kms_key_ring_iam_member`, and run `plan` and `apply`.
+
+## Creating K8s Resources
+
+Update `argocd/bootstrap-utilities/staging/values.yaml` or `argocd/bootstrap-utilities/production/values.yaml`
+to include the new shard. Under `rekor-tiles.shards`, copy the previous shard instantiation, and update
+the values accordingly based on the Terraform resources names. For `keyset`, copy the contents of
+`enc-keyset.cfg` (Since this value is encrypted, it's secure to include it in the config rather than
+use GCP Secrets).
+
+<!--- TODO: Add code pointers for values -->
+
+Merge, and wait for ArgoCD to create the resources and spin up the service.
```

Comment:
wait how long? is there some indicator? maybe an an oncall person would already know this?